### PR TITLE
Groups api

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: service/src/main/java/rdw/reporting/repository/GroupRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		this.jdbcTemplate = checkNotNull(jdbcTemplate);
+		this.getGroupSummaries = checkNotNull(getGroupSummaries);
+		this.getGroup = checkNotNull(getGroup);
+	}
+
+	public Set<GroupSummary> getGroupSummaries(@NotNull User user) {
+		return Sets.newHashSet(
+			jdbcTemplate.query(
+				getGroupSummaries,
+				ImmutableMap.of("user_login", "dwtest@example.com"),
```
Comment (malaffoon) [inline service/src/main/java/rdw/reporting/repository/GroupRepositoryImpl.java]: Can we push the hard-coding of the username up a layer or two? In other words, this line would be `ImmutableMap.of("user_login", user.getUsername()),` and the dwtest@example.com would be injected at the controller layer (and tests) as a hard-coded default.

File: service/src/main/java/rdw/reporting/repository/GroupRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		);
+	}
+
+	public Optional<Group> getGroup(@NotNull User user, long id) {
+		final ImmutableGroup.Builder group = ImmutableGroup.builder().id(id);
+		final List<Student> students = jdbcTemplate.query(
+			getGroup,
+			ImmutableMap.of(
+				"id", id,
+				"user_login", "dwtest@example.com"
```
Comment (malaffoon) [inline service/src/main/java/rdw/reporting/repository/GroupRepositoryImpl.java]: ditto

File: service/src/main/resources/application.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    getGroupSummaries:
+      >
+      select
+        reporting.roster.id as id,
+        reporting.roster.name as name,
+        count(*) as size
+      from reporting.student
+      join reporting.roster_membership on reporting.student.id=reporting.roster_membership.student_id
+      join reporting.roster on reporting.roster_membership.roster_id=reporting.roster.id
+      where reporting.roster.id in (select roster_id from reporting.user_roster where user_login=:user_login)
```
Comment (malaffoon) [inline service/src/main/resources/application.yaml]: Is the column in user_roster really `user_login`? Seems like it should just be `user_id` or `user_name`?

Comment (esotrope) [inline service/src/main/resources/application.yaml]: yes it currently is user_login

File: service/src/main/java/rdw/reporting/repository/GroupRepositoryImpl.java (hunk 1 above)
Comment (esotrope) [inline service/src/main/java/rdw/reporting/repository/GroupRepositoryImpl.java]: Actually I forgot to fix this - it should be local only

**Pull request merged**